### PR TITLE
Added better error handling, dotenvy for non-docker development

### PR DESCRIPTION
**Pull request by Nereuxofficial**
color-eyre is pretty awesome for good error messages and even some suggestions are provided to fix the error.

I took the liberty and added dotenvy which reads the .env file to environment variables, which should make development outside of a Docker container easier.

Currently the only place where it will not crash is sending a discord notification but idk if that's desireable

Also updated rusqlite cause why not

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -7,9 +7,12 @@ edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
+color-eyre = { version="0.6.2", default-features=false }
+# Allows to develop more easily when not in a docker container
+dotenvy = "0.15.7"
 openssl = { version="0.10.48", features=["vendored"] }
-reqwest = { version="0.11.14", features=["json"] }
-rusqlite = { version="0.28.0", features=["bundled"] }
+reqwest = { version="0.11.8", features=["json"] }
```
Comment (mfloto) [inline Cargo.toml]: could you change this to version `0.11.17`?

File: src/main.rs [hunk 2]
```diff
@@ -5,83 +5,89 @@ use std::env;
 
 use crate::discord::notify_discord_server;
 use crate::tm_response::{Offer, TMRes};
+use color_eyre::eyre::Context;
+use color_eyre::{Result, Report, Help};
+use core::fmt::Error;
```
Comment (mfloto) [inline src/main.rs]: Is the import `core::fmt::Error` ever used or can it be removed?

**Review by mfloto — CHANGES_REQUESTED** [review]
Thanks for the contribution. The console output looks great now. There are just two small things I noticed.

Don't worry about the Container-Action failing. It shouldn't even run on a PR.

File: src/main.rs (hunk 2 above)
Comment (Nereuxofficial) [reply src/main.rs]: Removed now, thanks

File: Cargo.toml [hunk 3]
```diff
@@ -11,7 +11,7 @@ color-eyre = { version="0.6.2", default-features=false }
 # Allows to develop more easily when not in a docker container
 dotenvy = "0.15.7"
 openssl = { version="0.10.48", features=["vendored"] }
-reqwest = { version="0.11.8", features=["json"] }
+reqwest = { version="0.11.7", features=["json"] }
```
Comment (mfloto) [inline Cargo.toml]: Just a typo here. The version should be `0.11.17`

**Review by mfloto — CHANGES_REQUESTED** [review]
Thanks again. There is just a small typo.

**Pull request merged**